File: Binary Tree To Array/BinaryTree.java
import java.util.ArrayList;

class BinaryTreeNode {
    int data;
    BinaryTreeNode left;
    BinaryTreeNode right;
    static int size = 0;

    BinaryTreeNode(int data) {
        this.data = data;
        ++size;
    }
}

public class BinaryTree {
    @SuppressWarnings("unused")
    private static void inorder(BinaryTreeNode node) {
        if (node != null) {
            inorder(node.left);
            System.out.print(node.data + " ");
            inorder(node.right);
        }
    }

    // Converting Binary Tree to a Array and preserving the formula
    private static ArrayList<Integer> binaryToArray(BinaryTreeNode node) {
        ArrayList<Integer> a = new ArrayList<Integer>();
        ArrayList<BinaryTreeNode> nodeList = new ArrayList<BinaryTreeNode>();
        nodeList.add(node);
        while (nodeList.size() > 0) {
            BinaryTreeNode temp = nodeList.get(0);
            nodeList.remove(0);

            if (temp != null) {
                a.add(temp.data);
                nodeList.add(temp.left);
                nodeList.add(temp.right);
            } else {
                a.add(null);
            }

        }
        // Removing extra null values from the array
        while (a.get(a.size() - 1) == null)
            a.remove(a.size() - 1);
        return a;
    }

    public static void main(String[] args) {
        BinaryTreeNode T = new BinaryTreeNode(20);
        T.left = new BinaryTreeNode(10);
        T.right = new BinaryTreeNode(30);
        T.right.left = new BinaryTreeNode(40);
        T.right.right = new BinaryTreeNode(5);
        // inorder(T);
        System.out.println();
        ArrayList<Integer> a = binaryToArray(T);
        for (Integer x : a) {
            System.out.print(x + " ");
        }
        System.out.println();

    }
}
// Formula for representing Binary Tree into Array
// if a node is at index: i
// its left child is at: 2 * i
// its right child is at: 2 * i + 1
// its parent is at: int (i/2)

File: Binary Tree To Array/BinaryTree.py
#!/usr/bin/python3
class BinaryTreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None


def inorder(node):
    if node:
        inorder(node.left)
        print(node.data, end=" ")
        inorder(node.right)


# Converting Binary Tree to a Array and preserving the formula
def binarytoArray(node):
    a = []
    q = []
    q.append(node)
    while len(q):
        temp = q.pop(0)
        if temp:
            a.append(temp.data)
            q.append(temp.left)
            q.append(temp.right)
        else:
            a.append(None)

    # Removing extra None values from the array
    while a[-1] == None:
        a.pop()
    return a


if __name__ == '__main__':
    T = BinaryTreeNode(20)
    T.left = BinaryTreeNode(10)
    T.right = BinaryTreeNode(30)
    T.right.left = BinaryTreeNode(40)
    T.right.right = BinaryTreeNode(5)
    inorder(T)
    print()
    print(binarytoArray(T))
    print()
# Formula for representing Binary Tree into Array
# if a node is at index: i
# its left child is at: 2 * i
# its right child is at: 2 * i + 1
# its parent is at: int (i/2)
